### ISSUE TITLE
Fail the build when `Dartfmt.dryRun` finds unformatted source?

### ISSUE THREAD

**Issue by seaneagan** [google/grinder.dart#210]
Currently `Dartfmt.dryRun` returns true if anything is not formatted.  Should we instead `fail("Unformatted source")` or similar in that case, rather than make the user do that by checking the return value?

**Comment by devoncarew** [comment]
@sethladd, thoughts?

If we're going to fail, we should make it clear to the callers. Perhaps adding an additional method called `failIfUnformatted()`? That could just call through to `dryRun()`.

**Comment by zoechi** [comment]
In my implementation I return the list of files which need to be formatted. If the list is not empty I can print the info and fail in the task or do some other action. An optional argument to fail automatically (default true) would be nice (for the simplest case)

**Comment by seaneagan** [comment]
```dart
class Formatter {
  void format(String glob);
  List<FileSystemEntity> getUnformatted(String glob);
  void verifyFormatted(String glob);
}
```

We should also consider calling the [programmatic API](https://github.com/dart-lang/dart_style#using-it-programmatically) rather than (or as an alternate option to) shelling out.

**Comment by zoechi** [comment]
Using the programmatic API needs a dependency even when the formatter packed with the SDK is used. I prefer shelling out because of that.

**Comment by seaneagan** [comment]
Yeah, I agree we should keep that option.  For a library-based solution I would definitely like to make it an optional dependency by pulling it in from another package.  Ideally [from dart_style itself](https://github.com/dart-lang/dart_style/issues/246).
